From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Sun, 9 Nov 2025 13:17:46 -0500
Subject: [PATCH] panel/plugin-manager: Don't free object owned by hashtable

When calling `g_hash_table_lookup`, the returned data is not owned by
the caller. We should ref it so we don't have to worry about freeing vs
not freeing later.

Ref https://github.com/BuddiesOfBudgie/budgie-desktop/issues/766

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 src/panel/plugin/plugin-manager.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/panel/plugin/plugin-manager.c b/src/panel/plugin/plugin-manager.c
index 49cb78b3..b6d93ad2 100644
--- a/src/panel/plugin/plugin-manager.c
+++ b/src/panel/plugin/plugin-manager.c
@@ -370,7 +370,10 @@ BudgieAppletInfo *budgie_panel_plugin_manager_load_applet_instance(BudgiePanelPl
 	}
 
 	plugin_name = g_settings_get_string(settings, BUDGIE_APPLET_KEY_NAME);
-	info = g_hash_table_lookup(self->plugins, plugin_name);
+
+	if (g_hash_table_contains(self->plugins, plugin_name)) {
+		info = g_object_ref(g_hash_table_lookup(self->plugins, plugin_name));
+	}
 
 	// Check if the plugin has been loaded
 	if (!PEAS_IS_PLUGIN_INFO(info)) {
